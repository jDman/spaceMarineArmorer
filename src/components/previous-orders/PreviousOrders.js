import React, { Fragment } from 'react';

import Button from '../button/Button';
import classes from './PreviousOrders.module.css';

const PreviousOrders = (props) => {
  const orderItemTypeClasses = {
    helmet: classes.PreviousOrdersListItemHelmet,
    body: classes.PreviousOrdersListItemBody,
    leg: classes.PreviousOrdersListItemLeg,
    arm: classes.PreviousOrdersListItemArm,
  };

  return (
    <div className={classes.PreviousOrders}>
      {!props.showPreviousOrders ? (
        <Button
          classes={classes.PreviousOrdersButton}
          clickHandler={() => props.showPreviousOrdersHandler(true)}
          buttonText="View previous orders"
        />
      ) : (
        <Button
          classes={classes.PreviousOrdersButton}
          clickHandler={() => props.showPreviousOrdersHandler(false)}
          buttonText="Hide previous orders"
        />
      )}

      {props.showPreviousOrders ? (
        <Fragment>
          {props.previousOrders.length ? (
            <Fragment>
              <h2 className={classes.PreviousOrdersHeading}>Previous orders</h2>
              <ul className={classes.PreviousOrders}>
                {props.previousOrders.map((order) => (
                  <div key={order._id}>
                    <div className={classes.PreviousOrdersListItemHeading}>
                      <h3>Order ID</h3>
                      <span>{order._id}</span>
                    </div>
                    {order.items.map((item) => (
                      <li
                        key={item.armor._id}
                        className={`${classes.PreviousOrdersListItem} ${
                          orderItemTypeClasses[item.armor.type]
                        }`}
                      >
                        <h4>{item.armor.name}</h4>
                        <p>Type {item.armor.type}</p>
                        <p>
                          Cost <small>{item.armor.cost}</small>
                        </p>
                      </li>
                    ))}
                  </div>
                ))}
              </ul>
            </Fragment>
          ) : (
            <h2>No orders found.</h2>
          )}
        </Fragment>
      ) : null}
    </div>
  );
};

export default PreviousOrders;
