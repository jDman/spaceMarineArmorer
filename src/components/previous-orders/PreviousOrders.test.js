import React from 'react';
import renderer from 'react-test-renderer';

import PreviousOrders from './PreviousOrders';

describe('<PreviousOrders />', () => {
  it('should render component and display list of previous orders', () => {
    const previousOrders = [
      {
        _id: '5e98106449355843575cf0c1',
        userId: '5e9435b51268b871ff874b8f',
        items: [
          {
            _id: '5e98106449355843575cf0c2',
            armor: {
              createdBy: {
                userId: '5e70dfb438cee83fd9e004fd',
                userName: 'Freddy',
              },
              stock: 4,
              shield: 10,
              discount: 10,
              _id: '5e96c3db8beefe1130ce80ff',
              type: 'body',
              cost: 200,
              protection: 'medium',
              quality: 'low',
              description: 'This is body armor...',
              company: 'adrax_corp',
              createdAt: '2020-04-15T08:20:43.170Z',
              updatedAt: '2020-04-15T08:20:43.170Z',
              __v: 0,
            },
            quantity: 2,
          },
          {
            _id: '5e98106449355843575cf0c3',
            armor: {
              createdBy: {
                userId: '5e70dfb438cee83fd9e004fd',
                userName: 'Freddy',
              },
              stock: 5,
              shield: 50,
              discount: 30,
              _id: '5e941bab2c9afa63aa5cadb3',
              type: 'helmet',
              cost: 200,
              protection: 'low',
              quality: 'low',
              description: 'Low tier trash',
              company: 'starscape_systems',
              createdAt: '2020-04-13T07:58:35.164Z',
              updatedAt: '2020-04-13T07:58:35.164Z',
              __v: 0,
            },
            quantity: 1,
          },
        ],
        totalCost: 600,
        createdAt: '2020-04-16T07:59:32.514Z',
        updatedAt: '2020-04-16T07:59:32.514Z',
        __v: 0,
      },
    ];

    const component = renderer.create(
      <PreviousOrders previousOrders={previousOrders} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
