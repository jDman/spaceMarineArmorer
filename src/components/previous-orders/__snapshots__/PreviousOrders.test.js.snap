// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviousOrders /> should render component and display list of previous orders 1`] = `
<div
  className="PreviousOrders"
>
  <button
    className="PreviousOrdersButton"
    onClick={[Function]}
    type="button"
  >
    View previous orders
  </button>
</div>
`;
